ui/src/provider/UserSettingsProvider.tsx

```diff
@@ -2,26 +2,40 @@ import * as React from 'react';
 import {useSettings} from '../gql/settings';
 import {CenteredSpinner} from '../common/CenteredSpinner';
 import moment, {LocaleSpecification} from 'moment';
-import {DateLocale, WeekDay} from '../gql/__generated__/globalTypes';
+import 'moment/locale/en-gb'; // DD/MM/YYYY but 24hr
+import 'moment/locale/en-au'; // DD/MM/YYYY but am/pm
+import {DateFormat, DateLocale, WeekDay} from '../gql/__generated__/globalTypes';
 import {expectNever} from '../utils/never';
 
-const setLocale = (locale: DateLocale, spec: LocaleSpecification) => {
+const setLocale = (locale: DateLocale, dateFormat: DateFormat, spec: LocaleSpecification) => {
     switch (locale) {
         case DateLocale.English:
-            moment.locale('en', spec);
+            if (dateFormat === DateFormat.MMDDYYYY) {
+                moment.locale('en', spec);
+            } else if (dateFormat === DateFormat.DDMMYYYY) {
+                moment.locale('en-au', spec);
+            } else {
+                throw new Error('Unexpected date format');
+            }
             return;
         case DateLocale.English24h:
-            moment.locale('en', {
-                ...spec,
-                longDateFormat: {
-                    LTS: 'HH:mm:ss',
-                    LT: 'HH:mm',
-                    L: 'MM/DD/YYYY',
-                    LL: 'MMMM D, YYYY',
-                    LLL: 'MMMM D, YYYY HH:mm',
-                    LLLL: 'dddd, MMMM D, YYYY HH:mm',
-                },
-            });
+            if (dateFormat === DateFormat.MMDDYYYY) {
+                moment.locale('en', {
+                    ...spec,
+                    longDateFormat: {
+                        LTS: 'HH:mm:ss',
+                        LT: 'HH:mm',
+                        L: 'MM/DD/YYYY',
+                        LL: 'MMMM D, YYYY',
+                        LLL: 'MMMM D, YYYY HH:mm',
+                        LLLL: 'dddd, MMMM D, YYYY HH:mm',
+                    },
+                });
+            } else if (dateFormat === DateFormat.DDMMYYYY) {
+                moment.locale('en-gb', spec);
```

Comment:
I dislike this, couldn't we add en-au & en-gb as locales, then we don't need another setting, and it is more explicit what the user gets?

Reply:
We totally can, but you had already implemented the custom 24hr english locale and I tried to minimize the breaking changes. Sounds like you're happy to allow this, I'll implement when I have some time. I'll try to find available locales to recreate the 4 + 1 potential permutations.